### PR TITLE
Muchocine - Spanish movie reviews dataset

### PR DESCRIPTION
**Pull request by mapmeld**

### CODE REVIEW

File: datasets/muchocine/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+annotations_creators: []
+language_creators: []
```
Comment (lhoestq) [inline datasets/muchocine/README.md:3]: can you fill those two fields as well ?

File: datasets/muchocine/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+- Dr. Fermín L. Cruz Mata
+
+### Licensing Information
+
+CC-BY-2.1
+
+### Citation Information
+
+See http://www.lsi.us.es/~fermin/index.php/Datasets
```
Comment (lhoestq) [inline datasets/muchocine/README.md:74]: looks like some sections are missing ?

Could you re-add them ?

You can leave their content with `[More Information Needed]`

File: datasets/muchocine/muchocine.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        )
+
+    def _split_generators(self, dl_manager):
+        my_urls = _URLs[self.config.name]
+        data_dir = dl_manager.download_and_extract(my_urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepaths": glob.glob(os.path.join(data_dir, "corpusCriticasCine", "*.xml")),
```
Comment (lhoestq) [inline datasets/muchocine/muchocine.py:66]:
```suggestion
                    "filepaths": sorted(glob.glob(os.path.join(data_dir, "corpusCriticasCine", "*.xml"))),
```
we need to make the order deterministic

File: datasets/muchocine/muchocine.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    def _generate_examples(self, filepaths, split):
+        for filepath in filepaths:
+            with open(filepath, encoding="latin-1") as f:
+                id = re.search(r"\d+\.xml", filepath)[0][:-4]
+
+                txt = f.read()
+                txt = txt.replace("&ldquo;", '"').replace("&rdquo;", '"').replace("&hellip;", "")
+                txt = txt.replace("&lsquo;", '"').replace("&rsquo;", '"').replace("&prime;", "")
+                txt = txt.replace("&agrave;", "à").replace("&ndash;", "-").replace("&egrave;", "è")
+                txt = txt.replace("&ouml;", "ö").replace("&ccedil;", "ç").replace("&", "and")
```
Comment (lhoestq) [inline datasets/muchocine/muchocine.py:81]: is there an encoding that works without having to do the replacements ?

**Review by lhoestq — COMMENTED** [review]
Looking pretty good so far ! Thanks for adding this one

I left a few comments

**Comment by lhoestq** [comment]
Hi @mapmeld !
have you had a chance to take a look at my suggestions ?

Feel free to ping me if you have questions or when you're ready for a review

**Comment by mapmeld** [comment]
@lhoestq unfortunately I don't have any more information about where the dataset comes from

File: datasets/muchocine/README.md (hunk 1 above)
Comment (lhoestq) [reply datasets/muchocine/README.md:3]: This should be 
```
annotations_creators:
- no-annotation
language_creators:
- found
```

**Comment by lhoestq** [comment]
It's fine, you can just add the sections titles back and leave the content with `[More Information Needed]`

**Comment by mapmeld** [comment]
added missing sections, updated the Python code ✅

**Review by lhoestq — APPROVED** [review]
Thank you !
LGTM

**Pull request merged**